CDP4Reporting/DataSource/ReportingDataSourceNode.cs

```diff
@@ -0,0 +1,288 @@
+                table.Columns.Add(hierarchy.FieldName, typeof(string));
+            }
+
+            foreach (var publicGetter in PublicGetters)
+            {
+                table.Columns.Add(publicGetter.Name, publicGetter.GetMethod.ReturnType);
+            }
+
+            return table;
+        }
+
+        /// <summary>
+        /// The <see cref="ReportingDataSourceRow"/> representation of the current node.
+        /// </summary>
+        private readonly T rowRepresentation;
+
+        /// <summary>
+        /// The parent node in the hierarhical tree upon which the data source is based.
+        /// </summary>
+        private readonly ReportingDataSourceNode<T> parent;
+
+        /// <summary>
+        /// The children nodes in the hierarhical tree upon which the data source is based.
+        /// </summary>
+        internal List<ReportingDataSourceNode<T>> Children { get; } = new List<ReportingDataSourceNode<T>>();
+
+        /// <summary>
+        /// The <see cref="CDP4Common.EngineeringModelData.NestedElement"/> associated with this node.
+        /// </summary>
+        internal readonly NestedElement NestedElement;
+
+        /// <summary>
+        /// The <see cref="CDP4Common.EngineeringModelData.ElementBase"/> associated with this node.
+        /// </summary>
+        internal ElementBase ElementBase =>
+            this.NestedElement.GetElementBase();
+
+        /// <summary>
+        /// The <see cref="CDP4Common.EngineeringModelData.ElementDefinition"/> representing this node.
+        /// </summary>
+        internal ElementDefinition ElementDefinition =>
+            this.NestedElement.GetElementDefinition();
+
+        /// <summary>
+        /// The <see cref="ElementUsage"/> representing this node, if it exists.
+        /// </summary>
+        internal ElementUsage ElementUsage =>
+            this.NestedElement.GetElementUsage();
+
+        /// <summary>
+        /// The filtering <see cref="CategoryHierarchy"/> that must be matched on the current <see cref="ElementBase"/>.
+        /// </summary>
+        private readonly CategoryHierarchy filterCategory;
+
+        /// <summary>
+        /// Gets the name of the field/column when the data is transfered to a <see cref="DataRow"/>.
+        /// </summary>
+        private string fieldName => this.filterCategory?.FieldName;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="ReportingDataSourceNode{T}"/> class.
+        /// </summary>
+        /// <param name="categoryHierarchy">
+        /// The <see cref="CategoryHierarchy"/> associated with this node's subtree.
+        /// </param>
+        /// <param name="topElement">
+        /// The <see cref="CDP4Common.EngineeringModelData.NestedElement"/> associated with this node.
+        /// </param>
+        /// <param name="parent">
+        /// The parent node in the hierarhical tree upon which the data source is based.
+        /// </param>
+        public ReportingDataSourceNode(
```

Comment:
They are now, but when extra functionality is added, the name NestedElementReportingDataSourceNode could be too specific.
Renaming of classes is a no-go, because classnames are already used in `existing` reports. 
We might be able to convert it, but that will be very tricky.

Reply:
I think the existing reports we gave to ESA were given as a test-drive, and not final, so we should be able to change them there (if need be) since we will provide some new reports as final.

Reply:
True, but I meant future changes when reports are being used in a live environment.

Reply:
This class' name stays generic.